src/utp_stream.cpp

```diff
@@ -1394,7 +1412,10 @@ bool utp_socket_impl::send_pkt(int const flags)
 		+ (sack ? sack + 2 : 0)
 		+ (close_reason ? 6 : 0);
 
-	int payload_size = std::min(m_write_buffer_size
+	// If the connection is finalizing we no longer want to include any payload
+	bool const finalizing = state() == state_t::fin_sent || (flags & pkt_fin) || m_eof;
```

Comment:
I don't think it's right to add `|| m_eof` at the end here. Just because the remote end is done sending payload doesn't mean we have sent all our payload yet.

Reply:
I agree. I'll remove it.